3rdParty/Lua/os-module-safety.patch

```diff
@@ -0,0 +1,23 @@
+diff '--color=auto' -Nru Lua-88246d621abf7b6fba9332f49229d507f020e450.orig/lua-5.4.6/src/loslib.c Lua-88246d621abf7b6fba9332f49229d507f020e450/lua-5.4.6/src/loslib.c
+--- Lua-88246d621abf7b6fba9332f49229d507f020e450.orig/lua-5.4.6/src/loslib.c	2023-11-04 19:06:10.234901751 +0100
++++ Lua-88246d621abf7b6fba9332f49229d507f020e450/lua-5.4.6/src/loslib.c	2023-11-04 19:06:45.459865904 +0100
```

Comment:
We can't patch it like this -- the user might be using system Lua.
To do this safely, one can create a sandbox environment, and only copy the safe functions into the os module https://gitlab.com/OpenMW/openmw/-/blob/c1f7a9c258fdebaf677579eab3a513f3c7e7271a/components/lua/luastate.cpp#L272

Reply:
I've modified `CMake/Dependencies.cmake` to always use the bundled Lua.

Reply:
We'll want a sandboxed environment eventually anyway, so I don't think this is a good idea.
See also my second point (I edited the comment) about us potentially wanting to use it for internal (trusted) Lua modules.

Reply:
Here is the exact line from OpenMW that restricts the OS packages in the sandbox:

```
        mCommonPackages["os"] = mSandboxEnv["os"]
            = makeReadOnly(tableFromPairs<std::string_view, sol::function>({ { "date", mSol["os"]["date"] },
                { "difftime", mSol["os"]["difftime"] }, { "time", mSol["os"]["time"] } }));
```

Reply:
Got it!

Reply:
> We'll want a sandboxed environment eventually anyway, so I don't think this is a good idea. See also my second point (I edited the comment) about us potentially wanting to use it for internal (trusted) Lua modules.

It's out of scope for me for sure, I'd need to dig into how Lua internals work to take a stab at this, especially after briefly looking at the OpenMW code :(

Reply:
No worries, we'll get around to it eventually for sure!